lumicks/pylake/fdcurve.py

```diff
@@ -99,6 +130,18 @@ def d(self):
             self._distance_cache = getattr(self, f"distance{self._primary_distance_channel}")
         return self._distance_cache
 
+    def sliced(self, force_min=-np.inf, force_max=np.inf, distance_min=0, distance_max=np.inf):
+        """Get a slice of the fd curve. Note that distances smaller or equal than zero are always omitted.
+
+        Parameters
+        ----------
+        force_min, force_max, distance_min, distance_max: float
+            Force and distance limits.
+        """
+        f, d = self.f.data, self.d.data
+        valid_idx = np.logical_and.reduce((d > 0, d >= distance_min, d < distance_max, f >= force_min, f < force_max))
+        return FDSlice(f[valid_idx], d[valid_idx])
```

Comment:
Why return this special `FDSlice`? You could just return a new `FDCurve`. See what `__getitem__()` is doing in this class. This method could be implemented in terms of `__getitem__()`:
```suggestion
        return self[valid_idx]
```

Reply:
This selection is not necessarily time-contiguous. `FDCurve`'s `__getitem__()` slices by time and always produces a temporally contiguous result. Also note that we are operating on `self.f.data` and `self.d.data` here, not `self.f` and `self.d`.

I mean, we have three options here.
- We also add handling for non-contiguous slicing to `FdCurve`. I don't really like this one since it would reduce the assumptions you can make when dealing with `FdCurve`'s quite a bit.
- We could consider enforcing time contiguous slices only when using this filter. The difference should be marginal for this use case. We'd have to be very careful how we document that though, since that would mean you're not actually slicing by force and distance, but finding the smallest subset of contiguous data that satisfies both. I'd be cool with looking into this, but it'd make the PR a bit bigger. Could probably use some of Ray's recent PR with `_sliced_by_distance` for this too.
- We leave it like this, but maybe rename the function to reflect what it is doing a bit better. `sliced_non_contiguous` or `sliced_mask`?

Reply:
I completely missed this caveat of `fd.sliced()`. So what's the actual motivation for having this as a public API? I can see that this PR is mainly about `FdEnsemble` and I can perfectly understand the motivation for that API. I can see `fd.sliced()` being used as an implementation detail of `FdEnsemble` but making it public API may be premature unless there is a real need for it.

I would vote for making `fd.sliced()` private for now and deliver only `FdEnsemble`. The sliced API is too strange right now and requires more work to make it a proper user-facing API. So if there's no real reason to make it public, I wouldn't.

Reply:
`_sliced` it is! :)